# Import Packages

In [216]:
import os #This will be used to determine where the file is stored
import pandas as pd 
import numpy as np

# Loading Data with Pandas

In [217]:
pwd = os.getcwd() #This will create a string of the folder this python script is stored in

In [218]:
filepath = pwd + '/Challenge_DataSet.xlsx'

In [219]:
orders_df = pd.read_excel(filepath)
orders_df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0       1.0     53.0    746.0         224.0          2.0           cash   
1       2.0     92.0    925.0          90.0          1.0           cash   
2       3.0     44.0    861.0         144.0          1.0           cash   
3       4.0     18.0    935.0         156.0          1.0    credit_card   
4       5.0     18.0    883.0         156.0          1.0    credit_card   

               created_at  
0 2017-03-13 12:36:56.190  
1 2017-03-03 17:38:51.999  
2 2017-03-14 04:23:55.595  
3 2017-03-26 12:43:36.649  
4 2017-03-01 04:35:10.773

# Data Exploration

In [220]:
orders_df['order_amount'].describe()
# Based on the analysis below we can see that:
    # The mean is being used to evaluate the AOV. 
    # The amounts range from 90 to 70k which means that our data is extremely spread out. 
    # That would explain the high std of 41k.


count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

In [226]:
np.median(orders_df['order_amount'])

284.0

# Determine Outliers

In [222]:
# We will determine the outliers by using the IRQ method.
Q1 = orders_df['order_amount'].quantile(0.25)
Q3 = orders_df['order_amount'].quantile(0.75)
Q1, Q3

IQR = Q3-Q1

In [223]:
#We will determine the upper and lower bounds
lower_limit = Q1 - 1.5*IQR
upper_limit= Q3 + 1.5*IQR
lower_limit, upper_limit

(-177.5, 730.5)

# Remove Outliers

In [224]:
#We will remove the outliers based on the calculated upper and lower limits. 
no_outliers_df = orders_df[(orders_df['order_amount'] < upper_limit) & (orders_df['order_amount'] > lower_limit)]

no_outliers_df.shape

(4859, 7)

In [225]:
no_outliers_df['order_amount'].describe()

count    4859.000000
mean      293.715374
std       144.453395
min        90.000000
25%       162.000000
50%       280.000000
75%       380.000000
max       730.000000
Name: order_amount, dtype: float64

# Summary of Analysis 

## Answers for Question #1: 

#### 1a. 
##### In this case, the mean ($3,145.13) was used to report the AOV. That is not the best metric to use for this dataset as there are many outliers/extreme values. The range of order amounts is $90 to $70k, which shows how widely distributed the data is.

#### 1b.  
##### Since there are outliers in the data, the median would be a more robust metric to use to evaluate the AOV. 
##### You can also use the mean if you remove the outliers. This can be done by using the Inter Quartile Range (IRQ) approach. IRQ is the difference between the 75th and 25th percentile and it can be used to find the upper and lower bounds of a dataset. Any values less than the lowerbound and greater than the upperbound will be your outliers. You can remove these outliers and determine the mean of the new dataset to get a more accurate AOV.  

#### 1c.  
##### If you evaluate the AOV using the median of the original dataset, the AOV would be $284. 
##### If you evaluate the AOV using the mean with outliers removed, the AOV would be $293.73.

 